fontconfig/src/lib.rs

```diff
@@ -326,6 +327,56 @@ impl<'fc> Drop for Pattern<'fc> {
     }
 }
 
+/// Wrapper around `FcStrList`
+pub struct StrList<'a> {
+    list: *mut sys::FcStrList,
+    _life: &'a PhantomData<()>
+}
+
+impl<'a> StrList<'a> {
+    unsafe fn from_raw(_: &Fontconfig, raw_list: *mut sys::FcStrSet) -> Self {
+        Self { list: raw_list, _life: &PhantomData }
+    }
+}
+
+impl<'a> Drop for StrList<'a> {
+    fn drop(&mut self) {
+        unsafe { ffi_dispatch!(LIB, FcStrListDone, self.list) };
+    }
+}
+
+impl<'a> Iterator for StrList<'a> {
+    type Item = &'a str;
+
+    fn next(&mut self) -> Option<&'a str> {
+        let lang_str: *mut sys::FcChar8 = unsafe { ffi_dispatch!(LIB, FcStrListNext, self.list) };
+        if lang_str.is_null() {
+            None
+        } else {
+            match unsafe { CStr::from_ptr(lang_str as *const c_char) }.to_str() {
```

Comment:
Here, there are multiple options that I see:
1. like implemented here, illegal strings are skipped; it's a bit obscure and silent but safer and more resilient
2. bluntly call `to_str().unwrap()` the cstr; less safe but the behavior is more straightforward and failures wont be swept under the rug
3. do not unwrap, instead return `Result<...>` and let the client code deal with failures to convert to Rust strings

I think 1. is the better option here but maybe there's a practice or convention that would require otherwise.

Reply:
Yeah I think 1 is probably a reasonable choice for now.